extensions/amp-brightcove/0.1/amp-brightcove.js

```diff
@@ -46,34 +46,59 @@ class AmpBrightcove extends AMP.BaseElement {
 
   /** @override */
   layoutCallback() {
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowfullscreen', 'true');
+    iframe.src = this.getIframeSrc_();
+    this.applyFillContent(iframe);
+    this.element.appendChild(iframe);
+    this.iframe_ = iframe;
+    return this.loadPromise(iframe);
+  }
+
+  /**
+   * @return {string}
+   * @private
+   */
+  getIframeSrc_() {
     const account = user().assert(
         this.element.getAttribute('data-account'),
         'The data-account attribute is required for <amp-brightcove> %s',
         this.element);
     const playerid = (this.element.getAttribute('data-player') ||
-      this.element.getAttribute('data-player-id') ||
-      'default');
+        this.element.getAttribute('data-player-id') ||
+        'default');
     const embed = (this.element.getAttribute('data-embed') || 'default');
-    const iframe = this.element.ownerDocument.createElement('iframe');
-    let src = `https://players.brightcove.net/${encodeURIComponent(account)}/${encodeURIComponent(playerid)}_${encodeURIComponent(embed)}/index.html`;
 
+    let src = `https://players.brightcove.net/${encodeURIComponent(account)}`
+        + `/${encodeURIComponent(playerid)}`
+        + `_${encodeURIComponent(embed)}/index.html`;
     if (this.element.getAttribute('data-playlist-id')) {
       src += '?playlistId=';
       src += this.encodeId_(this.element.getAttribute('data-playlist-id'));
     } else if (this.element.getAttribute('data-video-id')) {
       src += '?videoId=';
       src += this.encodeId_(this.element.getAttribute('data-video-id'));
     }
-
     // Pass through data-param-* attributes as params for plugin use
     src = addParamsToUrl(src, getDataParamsFromAttributes(this.element));
-    iframe.setAttribute('frameborder', '0');
-    iframe.setAttribute('allowfullscreen', 'true');
-    iframe.src = src;
-    this.applyFillContent(iframe);
-    this.element.appendChild(iframe);
-    this.iframe_ = iframe;
-    return this.loadPromise(iframe);
+    return src;
+  }
+
+  /** @override */
+  mutatedAttributesCallback(mutations) {
```

Comment:
Now we have a `user().assert` in the stack, which'll throw if the expression generates an empty string (probably others in the future will assert more complicated checks). Just want to make sure a thrown error here isn't going to break `#mutatedAttributesCallback` calls to other elements (some for loop gets aborted, etc).